src/profile/ProfileForm.tsx

```diff
@@ -37,585 +25,387 @@ interface ChangedUser {
+              <Form.Control
+                as="select"
+                onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+                  if (e.currentTarget.value === "editor") {
+                    props.context.setValue("is_editor", true);
+                  } else if (e.currentTarget.value === "contributor") {
+                    props.context.setValue("is_editor", false);
+                  }
                 }}
+                value={
+                  props.context.getValues().is_editor ? "editor" : "contributor"
+                }
+                className="custom-select"
+                custom
               >
-                Generate
-              </Button>
+                <option value="contributor">Contributor</option>
+                <option value="editor">Editor</option>
+              </Form.Control>
             </Col>
-            <Col>
-              <InputGroup>
-                <Form.Control
-                  type={showPassword ? "text" : "password"}
-                  name="password"
-                  onChange={onChange}
-                  isInvalid={
-                    state.errors.password && state.errors.password.length > 0
+          </Form.Group>
+        )}
+        {(props.user !== undefined && !auth.user?.is_editor) ||
+          (props.user?.username === auth.user?.username && (
+            <>
+              <h3>Change Password</h3>
+              <Form.Group as={Row} controlId="curPassword">
+                <Form.Label column sm={2}>
+                  Current Password
+                </Form.Label>
+                <Col sm={10}>
+                  <Field
+                    errors={props.context.errors}
+                    type="password"
+                    name="cur_password"
+                    ref={props.context.register({
+                      validate: (password: string) => {
+                        if (
+                          props.user !== undefined &&
+                          props.context.getValues().password &&
+                          !props.context.getValues().cur_password
+                        ) {
+                          return "USER.PASSWORD.CURRENT_REQUIRED";
+                        }
+                      },
+                    })}
+                  />
+                </Col>
+              </Form.Group>
+            </>
+          ))}
+        <Form.Group as={Form.Row} controlId="password">
+          <Form.Label column sm={2}>
+            New Password
+          </Form.Label>
+          <Col sm={2}>
+            <Button
+              className="w-100"
+              variant="secondary"
+              onClick={() => {
+                props.context.setValue("password", nanoid());
+                // set the password visible so you can see what you get
+                setShowPassword(true);
+              }}
+            >
+              Generate
+            </Button>
+          </Col>
+          <Col sm={8}>
+            <Field
```

Comment:
Extremely minor, but the password field should be marked as required.

Reply:
Are we requiring password entry for changing any field? I assumed it was just for password changing.

Reply:
Ack. Should've clarified --- it should only be required when creating a user.

Reply:
Oh, that makes sense yeah.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/6ff2b84ef452f865cbd12226bf238a1ade257d49